pymaker/__init__.py

```diff
@@ -592,13 +606,16 @@ async def transact_async(self, **kwargs) -> Optional[Receipt]:
             invocation was successful, or `None` if it failed.
         """
 
+        global next_nonce
         self.initial_time = time.time()
         unknown_kwargs = set(kwargs.keys()) - {'from_address', 'replace', 'gas', 'gas_buffer', 'gas_price'}
         if len(unknown_kwargs) > 0:
-            raise Exception(f"Unknown kwargs: {unknown_kwargs}")
+            raise ValueError(f"Unknown kwargs: {unknown_kwargs}")
 
-        # Get the from account.
+        # Get the from account; initialize the first nonce for the account.
         from_account = kwargs['from_address'].address if ('from_address' in kwargs) else self.web3.eth.defaultAccount
+        if not next_nonce or from_account not in next_nonce:
+            next_nonce[from_account] = self.web3.eth.getTransactionCount(from_account, block_identifier='pending')
```

Comment:
I may be misunderstanding the motivation for PR. In the context of `NonceCalculation.SERIAL`, if `next_nonce[from_account]` is initialized with `getTransactionCount`, couldn't we simply call it again, like we do in the `TX_COUNT` case?

Reply:
On these third-party node providers, `getTransactionCount` will only work for the first call.  Subsequent calls will return the same value until the transaction gets mined.  This is not the behavior observed when running a standalone node.

Reply:
Got it. That makes sense. 👍 

Only if convenient, consider adding `subsequent calls will return the same value until the transaction gets mined` or a similar line in the readme. I don't think it's particularly important but may amuse the curious reader.

Reply:
There's some nuance depending on node provider; I'm more comfortable with the current level of detail.

Reply:
Infura has load balancing, so if you send a tx to their endpoint and then immediately call `getTransactionCount(from_account, block_identifier='pending')` the call for pending tx's may reach another node that does not have the original tx in the pending pool, and you'll get back a number that's too small.

I will agree that this solution is somewhat inelegant, which is addressed in the README. It's effective and it works though. The only way I think that you'd be able to get around this is to implement some sort of redis+celery process manager that runs these transactions through serially and with a transaction lock. That's a lot more work than a counter though and may not be performant enough for what a keeper needs to do.